### ISSUE TITLE
Abuot FRAGMENT

### ISSUE THREAD

**Issue by soyleung** [bcosca/fatfree-core#298]
I see FRAGMENT is work in  $_SERVER['REQUEST_URI'],
$_SERVER['REQUEST_URI'] will no return FRAGMENT.
My project is work on nginx, I don't know other!
So,i think! FRAGMENT is unnecessary,more than what is due!!!

**Comment by KOTRET** [comment]
Yes, the fragment is not sent to the server by the clients (although they provide access to the fragment internally, see [FF](https://bugzilla.mozilla.org/show_bug.cgi?id=1264178), [Chrome](url)).
PHP's `parse_url` supports this, but as it is not part of `$_SERVER['REQUEST_URI'])`, there cannot be any FRAGMENT in the hive.

**Comment by ikkez** [comment]
I'm not quite sure why it's there. Perhaps for unit tests?

**Comment by KOTRET** [comment]
In these cases the REQUEST_URI must have been set to something specific, that imho would never apply to any real-world-scenario as the fragment is not sent by browsers and is never part of the REQUEST_URI. So this does not make any sense 😵

**Comment by ikkez** [comment]
I've removed it in v4
